airbyte-cdk/python/airbyte_cdk/sources/declarative/auth/oauth.py

```diff
@@ -45,102 +88,60 @@ def __init__(
         :param options: Additional runtime parameters to be used for string interpolation
         """
         self.config = config
-        self.token_refresh_endpoint = InterpolatedString.create(token_refresh_endpoint, options=options)
-        self.client_secret = InterpolatedString.create(client_secret, options=options)
-        self.client_id = InterpolatedString.create(client_id, options=options)
-        self.refresh_token = InterpolatedString.create(refresh_token, options=options)
+        self._token_refresh_endpoint = InterpolatedString.create(token_refresh_endpoint, options=options)
+        self._client_secret = InterpolatedString.create(client_secret, options=options)
+        self._client_id = InterpolatedString.create(client_id, options=options)
+        self._refresh_token = InterpolatedString.create(refresh_token, options=options)
         self.scopes = scopes
-        self.access_token_name = InterpolatedString.create(access_token_name, options=options)
-        self.expires_in_name = InterpolatedString.create(expires_in_name, options=options)
-        self.refresh_request_body = InterpolatedMapping(refresh_request_body or {}, options=options)
+        self._access_token_name = InterpolatedString.create(access_token_name, options=options)
+        self._expires_in_name = InterpolatedString.create(expires_in_name, options=options)
+        self._refresh_request_body = InterpolatedMapping(refresh_request_body or {}, options=options)
 
-        self.token_expiry_date = (
+        self._token_expiry_date = (
             pendulum.parse(InterpolatedString.create(token_expiry_date, options=options).eval(self.config))
             if token_expiry_date
             else pendulum.now().subtract(days=1)
         )
-        self.access_token = None
+        self._access_token = None
 
     @property
-    def config(self) -> Mapping[str, Any]:
-        return self._config
-
-    @config.setter
-    def config(self, value: Mapping[str, Any]):
-        self._config = value
+    def get_token_refresh_endpoint(self) -> str:
```

Comment:
We have to do away w/ the existing getter/setter pattern here, because the names will now conflict with the dataclass fields. This becomes a particular problem with type hints which will now resolve to the property methods instead of the actual type. I opted to rename the internal methods since we control the interface, otherwise all the field names have to get more verbose which directly affects the DX of the config.

One thing worth discussing is whether we drop the `@property` decorator. It used to be required when we were using getter/setter decorators, but now it doesn't really serve a purpose beyond allowing us to do `self.get_client_id` instead of `self.get_client_id()`.

Reply:
sounds like there's no point in having the property. +1 for removing it